routers/api/v1/repo/commits.go

```diff
@@ -195,6 +197,10 @@ func GetAllCommits(ctx *context.APIContext) {
 			// get commit specified by sha
 			baseCommit, err = ctx.Repo.GitRepo.GetCommit(sha)
 			if err != nil {
+				if git.IsErrNotExist(err) {
+					ctx.Status(http.StatusNoContent)
+					return
+				}
 				ctx.Error(http.StatusInternalServerError, "GetCommit", err)
 				return
```

Comment:
Two points:
1. > no commit found

Sounds like it should be `404` to me, not `204`
2. In that case, you can simplify this block to `ctx.NotFoundOrServerError`

Reply:
Yes, it should be 404 as it's no different than other endpoints return for not found user etc

Reply:
404 is return code when repository doesn't exist. But I can live with 404 too.